File: 01/solution.js
import {check, readInput} from "../Utils.js";

function part1(input) {
    return input[0].split('').map(a => {
        if (a === '(') return 1; else return -1
    }).sum()
}

function part2(input) {
    let floor = 0
    const chars = input[0].split('')
    for (let idx = 0; idx < chars.length; idx++) {
        if (chars[idx] === '(')
            floor++
        else
            floor--
        if (floor === -1)
            return idx + 1
    }
}

const testInput = readInput("test_input.txt")
const input = readInput("input.txt")

check(part1(testInput), -1)
console.log("Part 1: " + part1(input))

check(part2(testInput), 5)
console.log("Part 2: " + part2(input))

File: 01/solution.dart
import '../utils.dart';

dynamic part1(List<String> input) {
  return input.map(int.parse).sum;
}

dynamic part2(List<String> input) {
  final freqs = input.map(int.parse);
  final seen = {0};
  var current = 0;
  while (true) {
    for (var freq in freqs) {
      current += freq;
      if (seen.contains(current)) {
        return current;
      }
      seen.add(current);
    }
  }
}

main() {
  final input = readFile('input.txt');
  final testInput = readFile('test_input.txt');

  check(3, part1(testInput));
  print(part1(input));

  check(2, part2(testInput));
  print(part2(input));
}
